vis-cmds.c

```diff
@@ -632,42 +632,36 @@ static bool cmd_earlier_later(Vis *vis, Win *win, Command *cmd, const char *argv
 	return pos != EPOS;
 }
 
-static size_t space_replace(char *dest, const char *src, size_t maxlen) {
-	char *to = dest;
-	const char *from = src;
-	size_t destlen = 0;
-	size_t count = 0;
-	const size_t size = strlen("␣");
-	for (; *from != '\0' && destlen < maxlen - size - 1; from++) {
-		if (*from == ' ' && count < 3) {
-			strncpy(to, "␣", size);
-			count++;
-			destlen += size;
-			to += size;
+static size_t space_replace(char *dest, const char *src, size_t dlen) {
+	size_t i, invisiblebytecount = 0, size = strlen("␣");
+	for (i = 0; *src && i < dlen; src++) {
+		if (*src == ' ' && i < dlen - size - 1) {
+			memcpy(&dest[i], "␣", size);
+			i += size;
+			invisiblebytecount += 2;
```

Comment:
The `2` here should just be `size - 1`. Also the variable could just be called `invisiblebytes`.